core/pool.go

```diff
@@ -9,7 +9,7 @@ import (
 type Pool struct {
 }
 
-func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig) {
+func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig, fuse chan bool, ack chan bool) {
```

Comment:
I'm looking at some thread pool/goroutine pool articles, see if we don't have to spawn threads on the fly, but rather have a pool ready, and start task on the fly.